validation/citrus-validation-json/src/main/java/org/citrusframework/validation/json/JsonTextMessageValidator.java

```diff
@@ -56,190 +46,74 @@
-                                        jsonArrayControl.size(), jsonArrayReceived.size()));
-                        }
-                    }
-                    for (int i = 0; i < jsonArrayControl.size(); i++) {
-                        if (jsonArrayControl.get(i).getClass().isAssignableFrom(JSONObject.class)) {
-                            if (!jsonArrayReceived.get(i).getClass().isAssignableFrom(JSONObject.class)) {
-                                throw new ValidationException(ValidationUtils.buildValueMismatchErrorMessage("Value types not equal for entry: '" + controlKey + "'",
-                                            JSONObject.class.getName(), jsonArrayReceived.get(i).getClass().getName()));
-                            }
-
-                            validateJson(controlKey, (JSONObject) jsonArrayReceived.get(i),
-                                    (JSONObject) jsonArrayControl.get(i), validationContext, context, readContext);
-                        } else {
-                            if (!jsonArrayControl.get(i).equals(jsonArrayReceived.get(i))) {
-                                throw new ValidationException(ValidationUtils.buildValueMismatchErrorMessage("Values not equal for entry: '" + controlKey + "'",
-                                            jsonArrayControl.get(i), jsonArrayReceived.get(i)));
-                            }
-                        }
-                    }
-                } else {
-                    if (!controlValue.equals(receivedValue)) {
-                        throw new ValidationException(ValidationUtils.buildValueMismatchErrorMessage("Values not equal for entry: '" + controlKey + "'",
-                                    controlValue, receivedValue));
-                    }
-                }
-            } else if (ValidationMatcherUtils.isValidationMatcherExpression(controlValue.toString())) {
-                ValidationMatcherUtils.resolveValidationMatcher(controlKey,
-                        null,
-                        controlValue.toString(), context);
-            } else if (StringUtils.hasText(controlValue.toString())) {
-                throw new ValidationException(ValidationUtils.buildValueMismatchErrorMessage(
-                    "Values not equal for entry '" + controlKey + "'", controlValue.toString(), null));
-            }
-
-            if (logger.isDebugEnabled()) {
-                logger.debug("Validation successful for JSON entry '" + controlKey + "' (" + controlValue + ")");
-            }
-        }
+        var entryControl = new JsonElementValidatorItem<>(elementName, receivedJson, controlJson);
+        new JsonElementValidator(strict, context, validationContext.getIgnoreExpressions()).validate(entryControl);
     }
 
     /**
      * Checks if given element node is either on ignore list or
      * contains @ignore@ tag inside control message
+     * <p>
+     * TODO is that method still needed?
```

Comment:
It's nowhere used, but could still be a braking change - any thoughts?

Reply:
I think just deprecate it for removal and leave an `@deprecated use .. instead` comment. this way we can remove it within one of the next few releases.